assets/app/features/upgrade/controllers/upgrade-upgrade-administration-server.controller.spec.js

```diff
@@ -1,67 +1,172 @@
+                    errors: {}
+                },
+                nodes_services: {
+                    status: 'pending',
+                    errors: {}
+                },
+                nodes_db_dump: {
+                    status: 'pending',
+                    errors: {}
+                },
+                nodes_upgrade: {
+                    status: 'pending',
+                    errors: {}
+                },
+                finished: {
+                    status: 'pending',
+                    errors: {}
                 }
             }
         },
-        incompleteUpgradeResponse = {
-            data: {
-                version: '3.0',
-                addons: [],
-                upgrade: {
-                    upgrading: true,
-                    success: false,
-                    failed: false
+        incompleteUpgradeResponseData = {
+            current_step: 'admin_upgrade',
+            substep: null,
+            current_node: null,
+            steps: {
+                upgrade_prechecks: {
+                    status: 'pending',
+                    errors: {}
+                },
+                admin_backup: {
+                    status: 'pending',
+                    errors: {}
+                },
+                admin_repo_checks: {
+                    status: 'pending',
+                    errors: {}
+                },
+                admin_upgrade: {
+                    status: 'running',
+                    errors: {}
+                },
+                database: {
+                    status: 'pending',
+                    errors: {}
+                },
+                nodes_repo_checks: {
+                    status: 'pending',
+                    errors: {}
+                },
+                nodes_services: {
+                    status: 'pending',
+                    errors: {}
+                },
+                nodes_db_dump: {
+                    status: 'pending',
+                    errors: {}
+                },
+                nodes_upgrade: {
+                    status: 'pending',
+                    errors: {}
+                },
+                finished: {
+                    status: 'pending',
+                    errors: {}
                 }
             }
         },
-        initialResponse = {
-            version: '3.0',
-            addons: [],
-            upgrade: {
-                upgrading: false,
-                success: false,
-                failed: false
+        initialResponseData = {
+            current_step: 'admin_upgrade',
+            substep: null,
+            current_node: null,
+            steps: {
+                upgrade_prechecks: {
+                    status: 'pending',
+                    errors: {}
+                },
+                admin_backup: {
+                    status: 'pending',
+                    errors: {}
+                },
+                admin_repo_checks: {
+                    status: 'pending',
```

Comment:
Shouldn't `upgrade_prechecks`, `admin_backup` and `admin_repo_checks` be `passed` instead??
or current_step be set to upgrade_prechecks?